pyvista/__init__.py

```diff
@@ -74,3 +54,45 @@
 
 # Name used for unnamed scalars
 DEFAULT_SCALARS_NAME = 'Data'
+
+MAX_N_COLOR_BARS = 10
+
+
+# Lazily import/access the plotting module
+def __getattr__(name):
+    """Fetch an attribute ``name`` from ``globals()``or the ``pyvista.plotting`` module..
+
+    This override is implemented to prevent importing all of the plotting module
+    and GL-dependent VTK modules when importing PyVista.
+
+    Raises
+    ------
+    AttributeError
+        If the attribute is not found.
+
+    """
+    try:
+        return globals()[name]
+    except KeyError:
+        pass
+
+    import importlib
+    import inspect
+
+    whitelist = [
+        'demos',
+        'examples',
+        'ext',
+        'trame',
+        'utilities',
+    ]
+    if name in whitelist:
+        return importlib.import_module(f'pyvista.{name}')
+
+    _module = importlib.import_module('pyvista.plotting')
+    try:
+        feature = inspect.getattr_static(_module, name)
+    except AttributeError as e:
+        raise AttributeError(e) from None
```

Comment:
Isn't this mostly equivalent to
```suggestion
    try:
        feature = inspect.getattr_static(_module, name)
    except AttributeError:
        raise
```
Which is just
```suggestion
    inspect.getattr_static(_module, name)
```
Or is the point exactly to hide the `getattr_static` call from the traceback?

Reply:
> Or is the point exactly to hide the getattr_static call from the traceback?

This is what I was after.... but I didn't do a good job of this in hindsight. I'm improved the error message here so now this is what happens:

```py
>>> import pyvista as pv
>>> pv.Foo
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/local/KHQ/bane.sullivan/Software/pyvista/pyvista/pyvista/__init__.py", line 97, in __getattr__
    raise AttributeError(f"module 'pyvista' has no attribute '{name}'") from None
AttributeError: module 'pyvista' has no attribute 'Foo'
>>> 
```